runtime/stdlib/test.go

```diff
@@ -0,0 +1,471 @@
+	for i, parameter := range testContractType.ConstructorParameters {
+		result[i] = parameter.TypeAnnotation.Type
+	}
+	return result
+}()
+
+var blockchainBackendInterfaceType = func() *sema.InterfaceType {
+	typ, ok := testContractType.NestedTypes.Get(blockchainBackendTypeName)
+	if !ok {
+		panic(errors.NewUnexpectedError("cannot find type %s.%s", testContractTypeName, blockchainBackendTypeName))
+	}
+
+	interfaceType, ok := typ.(*sema.InterfaceType)
+	if !ok {
+		panic(errors.NewUnexpectedError("invalid type for %s. expected interface", blockchainBackendTypeName))
+	}
+
+	return interfaceType
+}()
+
+func init() {
+
+	// Enrich 'Test' contract with natively implemented functions
+	testContractType.Members.Set(
+		testAssertFunctionName,
+		sema.NewUnmeteredPublicFunctionMember(
+			testContractType,
+			testAssertFunctionName,
+			testAssertFunctionType,
+			testAssertFunctionDocString,
+		),
+	)
+	testContractType.Members.Set(
+		testNewEmulatorBlockchainFunctionName,
+		sema.NewUnmeteredPublicFunctionMember(
+			testContractType,
+			testNewEmulatorBlockchainFunctionName,
+			testNewEmulatorBlockchainFunctionType,
+			testNewEmulatorBlockchainFunctionDocString,
+		),
+	)
+
+	// Enrich 'Test' contract elaboration with natively implemented composite types.
+	// e.g: 'EmulatorBackend' type.
+	TestContractChecker.Elaboration.CompositeTypes[EmulatorBackendType.ID()] = EmulatorBackendType
+}
+
+var blockchainType = func() sema.Type {
+	typ, ok := testContractType.NestedTypes.Get(blockchainTypeName)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"cannot find type %s.%s",
+			testContractTypeName,
+			blockchainTypeName,
+		))
+	}
+
+	return typ
+}()
+
+// Functions belong to the 'Test' contract
+
+// 'Test.assert' function
+
+const testAssertFunctionDocString = `assert function of Test contract`
+
+const testAssertFunctionName = "assert"
+
+var testAssertFunctionType = &sema.FunctionType{
+	Parameters: []*sema.Parameter{
+		{
+			Label:      sema.ArgumentLabelNotRequired,
+			Identifier: "condition",
+			TypeAnnotation: sema.NewTypeAnnotation(
+				sema.BoolType,
+			),
+		},
+		{
+			Label:      sema.ArgumentLabelNotRequired,
+			Identifier: "message",
+			TypeAnnotation: sema.NewTypeAnnotation(
+				sema.StringType,
+			),
+		},
+	},
+	ReturnTypeAnnotation: sema.NewTypeAnnotation(
+		sema.VoidType,
+	),
+	RequiredArgumentCount: sema.RequiredArgumentCount(1),
+}
+
+var testAssertFunction = interpreter.NewUnmeteredHostFunctionValue(
+	func(invocation interpreter.Invocation) interpreter.Value {
+		condition, ok := invocation.Arguments[0].(interpreter.BoolValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		var message string
+		if len(invocation.Arguments) > 1 {
+			messageValue, ok := invocation.Arguments[1].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+			message = messageValue.Str
+		}
+
+		if !condition {
+			panic(AssertionError{
```

Comment:
This is out of scope for developing the test framework but do you plan for the framework to eventually support tests that continue after failure, collecting the errors along the way? I am thinking of Go's `assert` vs `require` for tests.

Reply:
Yea, we can iteratively add them. For now, this function is a sort of duplicate of the built-in `assert()` function and has the same behaviour. One remaining question is whether to keep both or remove one of them.